gcloud-java-examples/src/main/java/com/google/gcloud/examples/ResourceManagerExample.java

```diff
@@ -0,0 +1,147 @@
+      if (args.length > 0) {
+        String projectId = args[0];
+        ProjectInfo project = resourceManager.get(projectId);
+        if (project != null) {
+          System.out.printf(
+              "Successfully got project '%s': %s.%n", projectId, projectDetails(project));
+        } else {
+          System.out.printf("Could not find project '%s'.%n", projectId);
+        }
+      } else {
+        System.out.println(
+            "Error: must supply a project ID corresponding to a project for which you have viewing"
+            + " permissions. You can create a project and then call get using the same project ID"
+            + " if you have no other projects to use in this test.");
+      }
+    }
+
+    @Override
+    public String getRequiredParams() {
+      return "projectId";
+    }
+  }
+
+  private static class ListAction implements ResourceManagerAction {
+    @Override
+    public void run(ResourceManager resourceManager, String... args) {
+      System.out.println("Projects you can view:");
+      for (ProjectInfo project : resourceManager.list().values()) {
+        System.out.println(projectDetails(project));
+      }
+    }
+
+    @Override
+    public String getRequiredParams() {
+      return "";
+    }
+  }
+
+  static {
+    ACTIONS.put("create", new CreateAction());
+    ACTIONS.put("get", new GetAction());
+    ACTIONS.put("list", new ListAction());
+  }
+
+  private static String projectDetails(ProjectInfo project) {
+    return "{projectId:" + project.projectId() + ", projectNumber:" + project.projectNumber()
+        + ", createTimeMillis:" + project.createTimeMillis() + ", state:" + project.state() + "}";
+  }
+
+  public static void main(String... args) {
+    String actionName = args.length > 0 ? args[0].toLowerCase() : DEFAULT_ACTION;
+    ResourceManagerAction action = ACTIONS.get(actionName);
+    if (action == null) {
+      StringBuilder actionAndParams = new StringBuilder();
+      for (Map.Entry<String, ResourceManagerAction> entry : ACTIONS.entrySet()) {
+        actionAndParams.append(entry.getKey());
+        String param = entry.getValue().getRequiredParams();
+        if (param != null && !param.isEmpty()) {
+          actionAndParams.append(' ').append(param);
+        }
+        actionAndParams.append('|');
+      }
+      actionAndParams.setLength(actionAndParams.length() - 1);
+      System.out.printf(
+          "Usage: %s [%s]%n", ResourceManagerExample.class.getSimpleName(), actionAndParams);
+      return;
+    }
+
+    // If you want to access a local Resource Manager emulator (after creating and starting the
+    // LocalResourceManagerHelper), use the following code instead:
+    // ResourceManager resourceManager = LocalResourceManagerHelper.options().service();
+    ResourceManager resourceManager = ResourceManagerOptions.defaultInstance().service();
+    args = args.length > 1 ? Arrays.copyOfRange(args, 1, args.length) : new String[] {};
+    action.run(resourceManager, args);
```

Comment:
You could applied here a common validation for minimum number of args (based on getRequiredParams() - preferably returning array) and if mismatch display the usage for that action.